libraries/Microsoft.Bot.Builder.Ai/Translator.cs

```diff
@@ -89,8 +89,9 @@ public void AddNoTranslatePhrase(string noTranslatePhrase)
                     int trgLength = Int32.Parse(wordIndexes[1].Split(':')[1]) - trgstartIndex + 1; 
                     string trgWrd = trgMessage.Substring(trgstartIndex,trgLength);
                     int trgWrdINdex = Array.FindIndex(trgWrds, row => row == trgWrd);
-
-                    alignMap[srcWrdIndex] = trgWrdINdex;
+                    
+                    if(srcWrdIndex>0 && trgWrdINdex>0)
```

Comment:
it's ok. You can spell out "sourceWordIndex" and "targetWordIndex". :)

Reply:
changed the variable names